apps/perf-test-react-components/tasks/perf-test.ts

```diff
@@ -0,0 +1,330 @@
+//      - Add webpack serve option with aliasing?
+//    - Reference selection (local file, OUFR version, etc?)
+//    - Watch mode for flamegraphs.
+//      - Would require going back to webserve config mode?
+//  - Variance
+//    - Characterize variance
+//    - Verify results are repeatable and consistent
+//      - 1 tab vs. 100 tabs simulateneously
+//      - Eliminate or account for variance!
+//      - Minimize scenarios.
+//  - Further ideas:
+//    - Resizing page to determine reflow
+//    - React cascading updates on initial component render.
+//    - Monomorphic vs. Megamorphic Analysis:
+//      - Sean Larkin said that switching from polymorphic to monomorphic was a webpack optimization.
+//      - https://mrale.ph/blog/2015/01/11/whats-up-with-monomorphism.html
+//      - https://dzone.com/articles/impact-of-polymorphism-on-component-based-framewor
+
+// TODO: other args?
+// https://github.com/v8/v8/blob/master/src/flags/flag-definitions.h
+//  --log-timer-events
+//  --log-source-code
+
+// Analysis
+//  - Why is BaseComponent warnMutuallyExclusive appearing in flamegraphs?
+//    - It appears the CPU is being consumed simply by calling warnMututallyExclusive.
+//    - warnMutuallyExlusive impl is neutered but there still perf hit in setting up the args to call it.
+//    - The "get" in flamegraphs is caused by "this.className" arg.
+//    - makeAllSafe also consumes time just by having any component extend BaseComponent.
+//    - Puppeteer.tracing
+//      - Similar to using profiler in Chrome, does not show bottom-up analysis well
+//      - Seems to break V8 profile logging output.
+//        await page.tracing.start({ path: path.join(logPath, testLogFile[0] + '.trace') });
+//        await page.goto(testUrl);
+//        await page.tracing.stop();
+
+// Hardcoded PR deploy URL for local testing
+const DEPLOY_URL = 'fluentuipr.z22.web.core.windows.net';
+
+const urlForDeployPath = process.env.DEPLOYURL
+  ? `${process.env.DEPLOYURL}/perf-test-react-components`
+  : 'file://' + path.resolve(__dirname, '../dist/');
+
+// Temporarily comment out deploy site usage to speed up CI build time and support parallelization.
+// At some point perf test should be broken out from CI default pipeline entirely and then can go back to using deploy site.
+// For now, use local perf-test bundle so that perf-test job can run ASAP instead of waiting for the perf-test bundle to be deployed.
+// const urlForDeploy = urlForDeployPath + '/index.html';
+const urlForDeploy = 'file://' + path.resolve(__dirname, '../dist/') + '/index.html';
+
+const targetPath = `heads/${process.env.SYSTEM_PULLREQUEST_TARGETBRANCH || 'master'}`;
+// TODO: Once merged to master, replace /perf-test/index.html with /perf-test-react-components/index.html.
+// This is because we need /perf-test-react-components to be deployed to master first before being usable.
+const urlForMaster = `https://${process.env.DEPLOYHOST || DEPLOY_URL}/${targetPath}/perf-test/index.html`;
```

Comment:
Followup on TODO here after this PR is merged - need an initial deployment of the perf test results for this new package to the public deploy site first which will happen after this is merged